src/main/kotlin/com/exactpro/th2/conn/dirty/tcp/core/api/impl/Channel.kt

```diff
@@ -164,18 +165,28 @@ class Channel(
             val event = if (mode.mangle) mangler.onOutgoing(this@Channel, buffer, metadata) else null
             val messageId = nextMessageId(bookName, sessionGroup, sessionAlias, SECOND)
 
+            val data = Unpooled.copiedBuffer(buffer) // copied because buffer will be released after sending
             thenRunAsync({
-                if (mode.mangle) mangler.postOutgoing(this@Channel, buffer, metadata)
-                event?.run { storeEvent(messageID(messageId), eventId ?: this@Channel.eventId) }
-                onMessage.accept(buffer, messageId, metadata, eventId)
+                runCatching {
+                    logger.trace { "Post process of '${messageId.toJson()}' message id: ${hexDump(data)}" }
+                    if (mode.mangle) mangler.postOutgoing(this@Channel, buffer, metadata)
```

Comment:
I we are here, maybe we should also print the result of post-processing to the logs

Reply:
What do you mean? mangler.postOutgoing doesn't change buffer and return any result

Reply:
Let's forbid buffer change